### PR TITLE
Codecov reports for Android modules

### PR DESCRIPTION
**Pull request by buenaflor**
Ref: #1454 

#skip-changelog

### CODE REVIEW

**Comment by romtsn** [comment]
I'm a bit confused, because we're running test coverage on `src/test/java` and not `src/main/java`, though it should be the other way around.

image

Also, `sentry-compose` module is still missing from the coverage it looks like

**Comment by buenaflor** [comment]
@romtsn should be fixed now, also `sentry-compose` is also now included

**Comment by romtsn** [comment]
Looks better already, but still something is off. For example, `ActivityLifecycleIntegration` appears to be completely [uncovered](https://app.codecov.io/gh/getsentry/sentry-java/blob/chore%2Fcodecov-android/sentry-android-core%2Fsrc%2Fmain%2Fjava%2Fio%2Fsentry%2Fandroid%2Fcore%2FActivityLifecycleIntegration.java), although we have a pretty extensive test [covering it](https://github.com/getsentry/sentry-java/blob/main/sentry-android-core/src/test/java/io/sentry/android/core/ActivityLifecycleIntegrationTest.kt)

**Review by markushi — APPROVED** [review]

**Comment by buenaflor** [comment]
Seems to be resolved now, the percentage drops a bit but the coverage looks good so far

File: build.gradle.kts [hunk 1]
```diff
@@ -97,6 +98,32 @@ allprojects {
...
+        "sentry-android-navigation",
+        "sentry-android-ndk",
+        "sentry-android-okhttp",
+        "sentry-android-sqlite",
+        "sentry-android-timber",
+        "sentry-compose"
+    )
+    if (jacocoAndroidModules.contains(name)) {
+        afterEvaluate {
+            tasks.getByName("jacocoTestReport") {
```
Comment (romtsn) [inline build.gradle.kts:113]: you could also use the default Gradle's `Copy` task, which should be even pretty straightforward https://docs.gradle.org/current/dsl/org.gradle.api.tasks.Copy.html

**Comment by romtsn** [comment]
`sentry-compose` is missing again though, probably because the plugin is not aware of `androidMain`? 
image

**Comment by romtsn** [comment]
We might need to look into https://kotlin.github.io/kotlinx-kover/gradle-plugin/ for KMP modules

File: sentry-android-core/build.gradle.kts [hunk 2]
```diff
@@ -64,9 +65,14 @@ android {
     }
 }
 
+jacoco {
+    toolVersion = "0.8.10"
```
Comment (marandaneto) [inline sentry-android-core/build.gradle.kts:69]: M: Move it to Config.kt, DRY

File: sentry-android-core/build.gradle.kts [hunk 3]
```diff
@@ -64,9 +65,14 @@ android {
...
+}
+
 tasks.withType<Test> {
     configure<JacocoTaskExtension> {
-        isIncludeNoLocationClasses = false
+        isIncludeNoLocationClasses = true
+        excludes = listOf("jdk.internal.*")
```
Comment (marandaneto) [inline sentry-android-core/build.gradle.kts:75]: M: If its always the same, can also be moved to Config.kt

**Comment by buenaflor** [comment]
@romtsn 
`sentry-compose` is now included again, does the coverage look correct? (it shows similar results with Kover)

As for Kover: after testing it out, Codecov never could parse the kover.xml report apparently so I chose to implement it manually through jacoco with a custom task.

**Review by romtsn — APPROVED** [review]
LGTM 👍

**Pull request merged**